### DISCUSSION TITLE
Is it possible to use a list for sort in collectFile() ?

### DISCUSSION THREAD

**Discussion by ghost** [nextflow-io/nextflow#4342] [Q&A]
I know when using `collectFile()` I can specify a custom sort order using closure, but is it possible to use a list for this?

For example, if I have a tuple with an array of [fastq, md5] combinations, the order of the rows is messed up because Nextflow runs the processes asynchronously. When sorting this tuple by fastq names, I want to use the column of the csv file that was read in advance. 
Note that this order is not natural ordering and cannot be solved by fastq's name.

Like:
If I have `fastq_md5` like this, 
```
[b.fastq.gz, l3m4n5o6p7q8r9s0t1u2v3w4x5y6z7a8]
[c.fastq.gz, a0f3b1c2d3e4f5g6h7i8j9k0l1m2n3o4]
[d.fastq.gz, p5q6r7s8t9u0v1w2x3y4z5a6b7c8d9e0]
[a.fastq.gz, p5q6r7s8t9u0v1w2x3y4z5a6b7c8d9e0]
```
I want to sort `fastq_md5` with using `nameOrder` .
```
Channel.fromPath("fastq_name_list.csv").splitCsv(sep:",").map{row -> row[2]}.toList().set{nameOrder}
# nameOrder is like [d.fastq.gz, c.fastq.gz, a.fastq.gz, b.fastq.gz]
```
I know something is needed but I don't come up with.
```
fastq_md5.collectFile(name: "fastq_md5_sorted.txt", newLine: true, sort: { nameOrder.SomeKindOfProcessed() })
```

Thanks in advance.

**Comment by JAMKuttan** [comment]
+1

**Comment by bentsherman** [comment]
The sort option can be a closure

https://nextflow.io/docs/latest/reference/operator.html#collectfile
